CryptKeeperZK/crypt-keeper-extension
# feat: integrate `@cryptkeeperzk/semaphore-proof` package

## Explanation

This PR is for generating Semaphore proofs on Chrome browser platforms through using `Offscreen` feature, that's because of known bug issue in MV3 https://bugs.chromium.org/p/chromium/issues/detail?id=1219164
 
## More Information

* closes #590 
* closes #625 

## Screenshots/Screencaps

N/A

## Manual Testing Steps

1. Build extension using `pnpm run dev`
2. Run MockMerkle server `pnpm run merkle:start`
3. Run Demo `pnpm run demo:start`
4. Connect to the demo.
5. Generate a SemaphoreProof from demo. 
6. Expected to see the FullProof output in the demo page. 

![Screenshot_20230714_155039](https://github.com/CryptKeeperZK/crypt-keeper-extension/assets/36774944/441666a4-7914-4f50-90c7-8cdf4842ac53)


## Pre-Merge Checklist


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
